### ISSUE TITLE
[QUESTION]: Can this work with Bootstrap css?

### ISSUE THREAD

**Issue by simplecommerce** [pagedjs/pagedjs#205]
Hi,

I didn't see any post related to this so I was wondering, can this library work with bootstrap?

I am currently testing puppeteer to print an html template in PDF.
With bootstrap it works fine the layout is ok using the styles and the classes like flex.

But if I add this library in order to generate header/footer, etc.
The css breaks and no longer works properly.

I assume there are conflicts.
Is there a way to make them work together?

**Comment by Petercopter** [comment]
I use Bootstrap extensively, but I haven't used it with Paged.js. HOWEVER, we ARE using Tailwind CSS with Paged.js, so you're probably seeing similar issues we've had to overcome. I'm fairly confident Bootstrap should be usable.

The problems you're seeing are probably related to breakpoints. Have you tried working without breakpoint related Bootstrap classes? We had to create a new breakpoint because the print layout falls between the desktop and mobile. You might be seeing layout problems because it's trying to fall into a mobile layout, when you're expecting a desktop layout.

We are using the repeating header and footer functions provided by Paged.js, and we struggled, but we made it work.

Good luck!

**Comment by nbyloff** [comment]
I am using Bootstrap to render a bunch of long reports and have had success with this library except with tables that span multiple pages (lots of tickets/work going on around this area). But the latest beta-v1 version completely breaks Bootstrap and renders a blank screen. I imagine(hope) that will be fixed at some point. I don't use BS breakpoints but can confirm the non-beta versions work well with Bootstrap

**Comment by NigelCunningham** [comment]
Would you provide an example for me to test, please? Thanks!

**Comment by nbyloff** [comment]
@NigelCunningham Absolutely. See attached zip is HTML and some assets. I will say it doesn't render a blank screen unless a table is include. The first page is fine, but once it hits the table there's a JS error so it ends there. This does not happen in beta.0 or current paged.js release.

The actual JS error is:
````
Uncaught (in promise) TypeError: element is null
    findAllAncestors http://report.test/test_report.html:178
    afterPageLayout http://report.test/test_report.html:124
````

Which is here:
````
afterPageLayout(pageElement, page, breakToken, chunker) {
      this.chunker = chunker;
      this.splitTablesRefs = [];

      if (breakToken) {
        const node = breakToken.node;
        console.log(node);
        const tables = this.findAllAncestors(node, "table"); // <-- This line 
        if (node.tagName === "TABLE") {
          tables.push(node);
        }
...
...
...
}

findAllAncestors(element, selector) {
      const ancestors = [];
      while (element.parentNode && element.parentNode.nodeType === 1) { // element NULL here
        if (element.parentNode.matches(selector)) {
          ancestors.unshift(element.parentNode);
        }
        element = element.parentNode;
      }
      return ancestors;
    }
````

I only have two `class="page"` references in the example, but if you notice my `console.log(node);` line above, It prints 3 times when rendering. The third time it's null resulting in this error. 

The other issues I have I think are covered under other tickets and may be fixed already? (Header on one page, table starts another. Table rows are hidden/disappear when spanning across multiple pages). If you switch between beta.1 & beta.0 with this example you can see the header on a page by itself, but somewhat irrelevant to this issue.

Thanks for taking a look! 

[test_report.zip](https://github.com/user-attachments/files/15753288/test_report.zip)
